zerver/tests/test_scheduled_messages.py

```diff
@@ -656,13 +655,15 @@ def test_attachment_handling(self) -> None:
         attachment_file1 = StringIO("zulip!")
         attachment_file1.name = "dummy_1.txt"
         result = self.client_post("/json/user_uploads", {"file": attachment_file1})
-        path_id1 = re.sub(r"/user_uploads/", "", result.json()["url"])
+        assert result.json()["url"].startswith("/user_uploads/")
+        path_id1 = result.json()["url"].removeprefix("/user_uploads/")
```

Comment:
I feel like this pattern should do `url = result.json()["url"]` and then do the other steps on top of it?

Or maybe even better would be a test class method, like `self.extract_path_from_upload_url(result.json()["url"])`

@alexmv FYI.

Reply:
Yeah, there's absolutely a pattern to abstract out.